### DISCUSSION TITLE
Pretty-Print X-Model data

### DISCUSSION THREAD

**Discussion by vandre** [alpinejs/alpine#1970] [1. Help]
What would be the Alpine-way of formatting data like currency, dates or numbers in an edit form. Consider for example a form with a balance field. The current balance is stored in the backend as 123456.70, but I want to display it as $123,456.70. 

Another use case: for a phone number field automatically add the dashes as the user types.

I'm coming from an Angular background where you can use data pipes to transform how data is rendered regardless of how it is stored in the backend

**Comment by MuzafferDede** [comment]
You can do something like [this](https://codepen.io/muzafferdede/pen/QWvXzPX)

There are no any build in solutions but you could easily create your own using the api.
